android/tests_backend/widgets/textinput.py

```diff
@@ -38,6 +38,17 @@ def readonly(self):
         focusable_in_touch_mode = self.native.isFocusableInTouchMode()
         if focusable != focusable_in_touch_mode:
             raise ValueError(f"invalid state: {focusable=}, {focusable_in_touch_mode=}")
+
+        # Check if TYPE_TEXT_FLAG_NO_SUGGESTIONS is set in the input type
+        input_type = self.native.getInputType()
+        is_no_suggestions_set = (
+            input_type & InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS
+        ) != 0
+        if not is_no_suggestions_set:
+            raise ValueError(
+                "readonly is True, but TYPE_TEXT_FLAG_NO_SUGGESTIONS is not set."
+            )
```

Comment:
The `readonly` property is not *asserting* that the widget is readonly - it is returning the current readonly status (hence the `return not focusable`). This test is asserting that TYPE_TEXT_FLAG_NO_SUGGESTIONS is *always* set. 

```suggestion
        if input_type & InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS:
            # TYPE_TEXT_FLAG_NO_SUGGESTIONS is set
            if focusable:
                raise ValueError(
                    "TYPE_TEXT_FLAG_NO_SUGGESTIONS is not set on the input."
                )
        else:
            # TYPE_TEXT_FLAG_NO_SUGGESTIONS is not set
            if not focusable:
                raise ValueError(
                    "TYPE_TEXT_FLAG_NO_SUGGESTIONS has been set when the input is readonly."
                )
```